dvc/repo/experiments/queue/remove.py

```diff
@@ -1,38 +1,39 @@
-from typing import (
-    TYPE_CHECKING,
-    Collection,
-    Dict,
-    Iterable,
-    List,
-    Optional,
-    Set,
-    Union,
-)
+from typing import TYPE_CHECKING, Collection, Dict, Iterable, List, Set, Union
 
 from dvc.repo.experiments.exceptions import UnresolvedExpNamesError
 from dvc.repo.experiments.queue.base import QueueDoneResult
 
 if TYPE_CHECKING:
     from dvc.repo.experiments.queue.base import QueueEntry
-    from dvc.repo.experiments.queue.local import LocalCeleryQueue
+    from dvc.repo.experiments.queue.celery import LocalCeleryQueue
     from dvc.repo.experiments.stash import ExpStashEntry
 
 
-def _remove_queued_tasks(
+def remove_tasks(
     celery_queue: "LocalCeleryQueue",
-    queue_entries: Iterable[Optional["QueueEntry"]],
+    queue_entries: Iterable["QueueEntry"],
 ):
     """Remove tasks from task queue.
 
     Arguments:
-        queue_entries: An iterable list of queued task to remove
+        queue_entries: An iterable list of task to remove
     """
+    from celery.result import AsyncResult
+
     stash_revs: Dict[str, "ExpStashEntry"] = {}
+    failed_stash_revs: List["ExpStashEntry"] = []
+    done_entry_set: Set["QueueEntry"] = set()
     for entry in queue_entries:
-        if entry:
+        if entry.stash_rev in celery_queue.stash.stash_revs:
```

Comment:
`celery_queue.stash.stash_revs` should be cached in this function instead of being repeatedly accessed:

```py
queue_revs = celery_queue.stash.stash_revs
for entry in queue_entries:
    if entry.stash_rev in queue revs:
...
```

Otherwise the `stash.stash_revs` property will be re-computed each time it is accessed (which is expensive, and may also change)